app/main/views.py

```diff
@@ -182,7 +197,7 @@ def get_service(service_id):
     return jsonify(services=jsonify_service(service))
 
 
-@main.route('/archived-services/<int:archived_service_id>', methods=['GET'])
+@main.route('/archived-services/<string:archived_service_id>', methods=['GET'])
```

Comment:
This should still be integer, `archived_service_id` is table primary key, not `service_id`. And we're probably missing a test somewhere since this went unnoticed.